crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -957,15 +958,41 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                 pylint::rules::await_outside_async(checker, expr);
             }
         }
-        Expr::FString(ast::ExprFString { values, .. }) => {
+        Expr::FString(ast::ExprFString { elements, .. }) => {
             if checker.enabled(Rule::FStringMissingPlaceholders) {
-                pyflakes::rules::f_string_missing_placeholders(expr, values, checker);
+                pyflakes::rules::f_string_missing_placeholders(expr, elements, checker);
             }
             if checker.enabled(Rule::HardcodedSQLExpression) {
                 flake8_bandit::rules::hardcoded_sql_expression(checker, expr);
             }
             if checker.enabled(Rule::ExplicitFStringTypeConversion) {
-                ruff::rules::explicit_f_string_type_conversion(checker, expr, values);
+                ruff::rules::explicit_f_string_type_conversion(checker, expr, elements);
+            }
+            for element in elements {
+                if let ast::FStringElement::Literal(
+                    literal_element @ ast::FStringLiteralElement { value, range },
+                ) = element
+                {
+                    if checker.enabled(Rule::HardcodedBindAllInterfaces) {
+                        if let Some(diagnostic) =
+                            flake8_bandit::rules::hardcoded_bind_all_interfaces(value, *range)
```

Comment:
I would prefer to not pass in the range here as otherwise it becomes difficult to know which range is this within the rule function.

Regarding my other comment, I think you would want to pass in the f-string expression node by capturing it in the match case arm above on line 961:

```rust
        fstring @ Expr::FString(ast::ExprFString { elements, .. }) => {
```